### PR TITLE
OPSEXP-818:  revisit the way services are handled

### PR DESCRIPTION
**Pull request by nilsSure**
Work done
- Replaced usage of ```systemd``` with ```service``` module as is more general (supported init systems include BSD init, OpenRC, SysV, Solaris SMF, systemd, upstart. [docs](https://docs.ansible.com/ansible/2.9/modules/service_module.html#ansible-collections-ansible-builtin-service-module))
- created and/or renamed handlers that describe the state of a service when they are notified. all handlers have consistent names. Now ansible displays the states of the services at the end of each role. Now all handlers are notified and the logic of deciding the state of the service is in the handlers file. Example bellow:

```
- name: Add activemq.service
  template:
    src: activemq.service
    dest: /etc/systemd/system/activemq.service
    owner: "{{ username }}"
    group: "{{ group_name }}"
    mode: 'u=rwx,g=rwx,o=rx'
  notify:
    - get-services
    - enable-activemq
    - start-activemq
    - restart-activemq
```

```   
RUNNING HANDLER [../roles/activemq : get-services] ***********************************************************************************************************************************************************************************************************************************************************************************************************************
ok: [activemq_1]

RUNNING HANDLER [../roles/activemq : enable-activemq] ********************************************************************************************************************************************************************************************************************************************************************************************************************
changed: [activemq_1]

RUNNING HANDLER [../roles/activemq : start-activemq] *********************************************************************************************************************************************************************************************************************************************************************************************************************
changed: [activemq_1]

RUNNING HANDLER [../roles/activemq : restart-activemq] *******************************************************************************************************************************************************************************************************************************************************************************************************************
skipping: [activemq_1]
``` 

- as per [OPSEXP-740](https://alfresco.atlassian.net/browse/OPSEXP-740) I changed the download location of activemq archive to ```/tmp/ansible_artefacts/``` . added ```idempotence``` test to molecule test sequence
- simplified the logic that writes env variables in setenv file for ```transformers```, ```sfs``` and ```trouter``` roles
- fixed ```search```, ```sfs```, ```tomcat```, ```trouter```  roles idempotency.

### CODE REVIEW

**Comment by alxgomz** [comment]
I'm not sure I understand  the purpose of the get-services handlers.

File: roles/adw/handlers/main.yml [hunk 1]
```diff
@@ -1,13 +1,43 @@
 ---
 # handlers file for adw
-- name: validate nginx configuration
-  command: nginx -t -c /etc/nginx/nginx.conf
-  changed_when: false
-- name: reload nginx
-  service: name=nginx state=reloaded
+# - name: validate nginx configuration
```
Comment (sergiuv2020) [inline roles/adw/handlers/main.yml:3]: why did you leave this here?

**Review by sergiuv2020 — APPROVED** [review]

File: roles/adw/handlers/main.yml (hunk 1 above)
Comment (nilsSure) [reply roles/adw/handlers/main.yml:3]: wasn't sure of the purpose of this handler and just commented it as it is not notified anywhere. Will remove it

Comment (nilsSure) [reply roles/adw/handlers/main.yml]: removed

**Review by gavincornwell — APPROVED** [review]
It's great to see consistency, also see a green build so LGTM

**Review by alxgomz — APPROVED** [review]

File: roles/activemq/handlers/main.yml [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+- name: get-services
+  service_facts:
+  register: services_state
+
+- name: enable-activemq
+  service:
+    name: activemq
+    enabled: true
+    daemon_reload: true
+  when: services_state.ansible_facts.services['activemq.service'].status != 'enabled'
```
[the remark this answers is not in the document]
Comment (nilsSure) [reply roles/activemq/handlers/main.yml:12]: I'm using this mechanism to display the state of the service at the of the role. All these tasks can be done in one go with  something similar to this:

```
- name: start-activemq
  service:
    name: activemq
    state: restarted
    enabled: true
    daemon_reload: true
```
I'd rather have more self explanatory code, than having fewer lines that don't give off much info about what they are doing

File: roles/repository/handlers/main.yml [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
+    port: "{{ repo_db_port }}"
+  register: go_signal
+
+- name: enable-alfresco-content
+  service:
     name: "alfresco-content"
-    state: restarted
+    enabled: true
     daemon_reload: true
+  when: not go_signal.failed and services_state.ansible_facts.services['alfresco-content.service'].status != 'enabled'
```
[the remark this answers is not in the document]
Comment (nilsSure) [reply roles/repository/handlers/main.yml:26]: There is no point in enabling/starting alfresco-content.service if there is no database. 
Don't know why this doesn't show here, but there is a condition for when to run the "wait-for-db" handler

```
- name: wait-for-db
  wait_for:
    host: "{{ db_host }}"
    port: "{{ repo_db_port }}"
  register: go_signal
  when: repo_db_url == ""

- name: set-go-signal
  set_fact:
    go_signal: {
      'failed': false
    }
  when: repo_db_url != ""

- name: enable-alfresco-content
  service:
    name: "alfresco-content"
    enabled: true
    daemon_reload: true
  when: not go_signal.failed and services_state.ansible_facts.services['alfresco-content.service'].status != 'enabled'
```

**Pull request merged**